client/components/Settings.js

```diff
@@ -15,20 +15,48 @@ import CustomBankField from './CustomBankField';
 import {OpCatChartTypeSelect, OpCatChartPeriodSelect} from './Charts';
 import T from './Translated';
 
-
 class Account extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDefault: (store.getDefaultAccountId() === this.props.account.id)
+        };
+    }
+
+    componentDidMount() {
+        store.on(State.defaultAccount, function() {
+            this.setState({isDefault: (store.getDefaultAccountId() === this.props.account.id)});
+        }.bind(this));
```

Comment:
style nit: if you use an arrow function `() =>` you can remove  the `.bind(this)`